### PR TITLE
initial commit of prefix-conversions

### PR DESCRIPTION
**Pull request by lance-pyles**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: conversions/prefix_conversions.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    1000
+    >>> convert_si_prefix(1, SI_Unit.mega, SI_Unit.giga)
+    0.001
+    >>> convert_si_prefix(1, SI_Unit.kilo, SI_Unit.kilo)
+    1
+    >>> convert_si_prefix(1, 'giga', 'mega')
+    1000
+    """
+
+    if type(known_prefix) is str:
```
Comment (cclauss) [inline conversions/prefix_conversions.py:62]:
```suggestion
    if isinstance(known_prefix, str):
```
See PEP8

File: conversions/prefix_conversions.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+        known_prefix: SI_Unit = SI_Unit[known_prefix]
+
+    if type(unknown_prefix) is str:
```
Comment (cclauss) [inline conversions/prefix_conversions.py:66]: isinstance()

File: conversions/prefix_conversions.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    0.001
+    >>> convert_si_prefix(1, SI_Unit.kilo, SI_Unit.kilo)
+    1
+    >>> convert_si_prefix(1, 'giga', 'mega')
+    1000
+    """
+
+    if type(known_prefix) is str:
+
+        known_prefix: SI_Unit = SI_Unit[known_prefix]
```
Comment (cclauss) [inline conversions/prefix_conversions.py:64]:
```suggestion
        known_prefix: SI_Unit = SI_Unit[known_prefix.lower()]
```
and please add a test that uses mixed case or upper case.

File: conversions/prefix_conversions.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
+"""
+Convert SI and Binary prefixes
```
Comment (cclauss) [inline conversions/prefix_conversions.py:2]: Does SI stand for System Integrator or Serial Interface and why does the reader have to guess?

Please add the URL https://en.wikipedia.org/wiki/International_System_of_Units or similar to help the reader.

File: conversions/prefix_conversions.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+"""
+
+from enum import Enum
+from typing import Union
+
+
+class SI_Unit(Enum):
+
```
Comment (cclauss) [inline conversions/prefix_conversions.py:10]:
```suggestion
```
Please lose these blank lines and the blank lines inside your functions.  They force the reader to do a lot of scrolling and they do not help with readability/understandability of the code.

Comment (lance-pyles) [reply conversions/prefix_conversions.py:10]: when I black the file it adds those two blank lines above class SI_Unit

**Pull request merged**